control/activate_venv.sh

```diff
@@ -35,7 +35,9 @@ fi
 
 # activate virtualenv if it exists, otherwise exit with error
 if [ -f "$VENV/bin/activate" ]; then
-    source "$VENV/bin/activate"
+    if [ $( which python ) != "$VENV/bin/python" ]; then
```

Comment:
I'm a little nervous about this technique since it's dependent on implementation details of the virtualenv provider. For example, I did a similar test in my dev environment:
```sh
$ which python
/home/dimagi/.pyenv/shims/python
$ echo "$VIRTUAL_ENV/bin/python"
/home/dimagi/.pyenv/versions/3.9.12/envs/cchq/bin/python
```
I think typical way to tell if a virtualenv is activated is to test if `$VIRTUAL_ENV` has a value. Alternately, you can just go ahead and `source "$VENV/bin/activate"` unconditionally—activating twice should be fine.

Reply:
Yeah I misunderstood an issue and chalked it up to this, but you can indeed attempt to activate an active virtualenv without issues. See 343a0d014be3826eb8692b5fc957eb8484cb2fc0